### PR TITLE
Replace search system with @docsearch/react

### PR DESCRIPTION
**Pull request by GabLeRoux**
#### Changes

* Replace search system with @docsearch/react
* Removed extraction code

- [x] Fix the pipeline (it works locally)
- [x] Integrate `@docsearch/react`
- [ ] Remove legacy indexing commands (see `tools/search/`)
- [ ] Determine where _selector config_ must be specified. This used to be a [config file](https://docsearch.algolia.com/docs/config-file) and it is now configured on the crawler itself. Can it be hosted in the repository? So far, it seems it can only be set in [https://crawler.algolia.com/admin/](https://crawler.algolia.com/admin/)
- [ ] Trigger crawler on a per-branch basis using `re-index` api call:  
  https://www.algolia.com/doc/rest-api/crawler/#reindex-with-a-crawler
- [ ] Use a prefix for branches (like we currently do). See `indexPrefix` in [`create_crawler.json` in Create a crawler](https://www.algolia.com/doc/rest-api/crawler/?client=json#create-a-crawler))
- [ ] Add a sitemap (this gets automatically crawled)

## Draft

```
- get crawlers
- check if prefix from branch name exists (see slugify or something in our code)
    - if exists, select that crawler
    - if not exists: create crawler (id gets returned)
- use crawler to index
- poll until success on all pages from "count"
```

## initial notes

I wonder if it's better or not. It sure looks good and it's user friendly, but I would expect to reach the unity builder docs when searching for `builder`. We might want to have a look at these useful links:

DocSearch examples: https://github.com/algolia/docsearch/tree/next/examples
CodeSandBox for React: https://codesandbox.io/s/docsearch-react-v3-playground-619yg
CodeSandBox for Vanilla JavaScript: https://codesandbox.io/s/docsearchjs-v3-playground-z9oxj
**DocSearch documentation: https://docsearch.algolia.com/**

For example, it might be necessary to provide a config file as shown here:  
https://docsearch.algolia.com/docs/config-file

Edit: This is already legacy, configuration moved to the crawlers tool.

Could be interesting to improve categorization of provided results.

#### Checklist

- [x] Read the contribution [guide](../blob/main/CONTRIBUTING.md) and accept the [code](../blob/main/CODE_OF_CONDUCT.md) of conduct
- [x] Readme (updated or not needed)
- [x] Tests (added, updated or not needed)

### CODE REVIEW

File: components/layout/header/search-bar.tsx [hunk 1]
```diff
@@ -1,56 +1,18 @@
...
-          <Stats />
-          <span>
-            Powered by
-            <a href="http://algolia.com/">
-              <SiAlgolia size={24} style={{ marginLeft: 8 }} />
-            </a>
-          </span>
-        </div>
-      </div>
+      <DocSearch apiKey="d085b783b0341babb98dc7fdc6a2a8be" indexName="game" />
```
Comment (webbertakken) [inline components/layout/header/search-bar.tsx:15]: 👀

File: components/layout/header/search-bar.tsx [hunk 2]
```diff
@@ -1,56 +1,18 @@
-import React, { useState } from 'react';
-import { SiAlgolia } from 'react-icons/all';
-import { SearchBox, Pagination } from 'react-instantsearch-dom';
-import Configuration from '@/components/search/configuration';
-import SearchResults from '@/components/search/results';
-import Stats from '@/components/search/stats';
+import React from 'react';
```
Comment (webbertakken) [inline components/layout/header/search-bar.tsx:1]: You don't have to import react in every file. Next already takes care of that.

File: components/layout/header/search-bar.tsx (hunk 1 above)
Comment (GabLeRoux) [reply components/layout/header/search-bar.tsx:15]: from what I understand, this api key is already public. I've seen many of them in the wild, here's an example:  
https://github.com/DataDog/documentation/blob/master/src/scripts/config/config-docs.js

I think they know what they're doing 🤞

**Comment by webbertakken** [comment]
#### 📝 Changed routes:


| [**…...documentation-page\]**](https://website-c6um9gv18-game-ci.vercel.app/docs/[...documentation-page]) |
|:-:|
|<a href="https://website-c6um9gv18-game-ci.vercel.app/docs/[...documentation-page]">Screenshot of /docs/[...documentation-page]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/ej_gBW_jXZMyogxTa841kF1T3UILs-BLdAAvwDBfw-Iu1vV1bzxjwWDZwzFETmTby-_Kd96g_Pvrtei3WlRf1g.png">(view full size)</a> |







Commit 9ed99cfc01aee515293d5255d73858c65e0e1d9c (https://website-c6um9gv18-game-ci.vercel.app).

**Comment by GabLeRoux** [comment]
> #### 📝 Changed routes:
> [**…...documentation-page]**](https://website-c6um9gv18-game-ci.vercel.app/docs/%5B...documentation-page%5D)
> [Screenshot of /docs/[...documentation-page]](https://website-c6um9gv18-game-ci.vercel.app/docs/[...documentation-page])
> [(view full size)](https://images.weserv.nl?url=https://sjc.microlink.io/ej_gBW_jXZMyogxTa841kF1T3UILs-BLdAAvwDBfw-Iu1vV1bzxjwWDZwzFETmTby-_Kd96g_Pvrtei3WlRf1g.png)
> 
> Commit [9ed99cf](https://github.com/game-ci/documentation/commit/9ed99cfc01aee515293d5255d73858c65e0e1d9c) ([website-c6um9gv18-game-ci.vercel.app](https://website-c6um9gv18-game-ci.vercel.app)).

Did I break something while deleting some code? 🥶

**Comment by webbertakken** [comment]
> > #### 📝 Changed routes:
> > [**…...documentation-page]**](https://website-c6um9gv18-game-ci.vercel.app/docs/%5B...documentation-page%5D)
> > [Screenshot of /docs/[...documentation-page]](https://website-c6um9gv18-game-ci.vercel.app/docs/%5B...documentation-page%5D)
> > [(view full size)](https://images.weserv.nl?url=https://sjc.microlink.io/ej_gBW_jXZMyogxTa841kF1T3UILs-BLdAAvwDBfw-Iu1vV1bzxjwWDZwzFETmTby-_Kd96g_Pvrtei3WlRf1g.png)
> > Commit [9ed99cf](https://github.com/game-ci/documentation/commit/9ed99cfc01aee515293d5255d73858c65e0e1d9c) ([website-c6um9gv18-game-ci.vercel.app](https://website-c6um9gv18-game-ci.vercel.app)).
> 
> Did I break something while deleting some code? 🥶

Not sure. I was a robot there

**Comment by GabLeRoux** [comment]
Ahah yeah I know about the bot :P, just wondering if it's normal it says it changed.

So for the search always pointing to prod, I agree it's a downside, but on the other hand, much less code to maintain so it's a trade-off.

I tried it to explore the solution, still unsure if it's better.

Maybe there's an in-between where we use the component from docsearch, but connect it to the self managed index.

Personally I really like how it's convenient with the search shortcut and how it looks.

**Comment by webbertakken** [comment]
I'm probably biased as I put 2 weekends into properly creating the indices and implementing the search and making it work through CI for all branches.

If you say it's worth it then I believe you 😆

**Comment by webbertakken** [comment]
If it's not convincing, shall we keep the more robust system that we already have, which also works per-branch and has more customisability?

Or is it better to move forward with this?

**Comment by GabLeRoux** [comment]
I think we need to put some more time in this as the quick integration I tried is not as good as what we currently have.

I would say the main blocker right now is the part where I removed the config that tells what to index and how.

I don't exactly know how we could achieve this, but I really think the perfect solution would be to do exactly [as I described here](https://github.com/game-ci/documentation/pull/218#issuecomment-918378850)

From this draft pr, I think the most important thing is `@docsearch/react` component is documented here:  
https://autocomplete-experimental.netlify.app/docs/docsearch/

There's probably a way to just use that with the index we're building on our own so we wouldn't have to maintain a custom component with pagination, etc.

**Review by webbertakken — APPROVED** [review]
I believe this newer docsearch might indeed be the way to go. Just need proper integration and remove the old code.

**Pull request closed without merging**